### PR TITLE
OP-1166 Fix lot validation

### PR DESCRIPTION
**Pull request by mwithi**
See OP-1166.

### CODE REVIEW

File: src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java [hunk 1]
```diff
@@ -112,42 +112,41 @@ protected void validateMovement(Movement movement, boolean checkReference) throw
...
+			}
 
-				if (lot.getPreparationDate() != null && lot.getDueDate() != null && lot.getPreparationDate().isAfter(lot.getDueDate())) {
-					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thepreparationdatecannotbyaftertheduedate.msg")));
-				}
+			if (lot.getPreparationDate() != null && lot.getDueDate() != null && lot.getPreparationDate().isAfter(lot.getDueDate())) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thepreparationdatecannotbyaftertheduedate.msg")));
 			}
+		}
+		List<Integer> medicalIds = ioOperations.getMedicalsFromLot(lot.getCode());
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java:134]: Shouldn't this be inside the `if (lot != null)` check?  If `lot` is `null` then `log.getCode()` will cause a NPE.

**Review by dbmalkovsky — COMMENTED** [review]
Added a question after review of the code.   Other than that I believe it looks OK

File: src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java:134]: Good catch, all of them indeed... let me revise the logic. Thanks!

Comment (mwithi) [reply src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java]: @dbmalkovsky please check now. Added also https://github.com/informatici/openhospital-gui/pull/1936. Thanks!

**Review by dbmalkovsky — APPROVED** [review]
LGTM

**Pull request merged**